New digit label handling for ITSMFT

sawenzel opened the pull request:
This PR achieves the following breakthroughs:

a) Ability to digitize a full digit time frame including MC labels for ITS.
b) Spike memory reduction by a factor of ~2 for the ITS digitizer workflow.

This is achieved through:

a) Application of the new const + view-only label containers in digitization and reconstruction. This application achieves
almost zero-copy treatment of labels.
b) A demonstrator showing how to do zero-copy writing to a ROOT tree (directly streaming from the buffers in shared memory) in the ITSMFT digit writer. This demonstrator may be interesting for application inside the RootTreeWriter in general.

This PR needs merging of #4411 first. Since data formats are changed, a few other places (such as macros) need to be adapted to the new label container. @shahor02 @iouribelikov : Please take a look and let me know your opinion.

In future, other things such as cluster labels can be converted in the same spirit.

shahor02 commented on Detectors/ITSMFT/common/workflow/src/DigitWriterSpec.cxx:
```diff
@@ -41,15 +72,95 @@ DataProcessorSpec getDigitWriterSpec(bool mctruth, o2::header::DataOrigin detOri
+              // but immediately adopt the views in standard std::vectors *** THIS IS AN INTERNAL HACK ***
+              if (mctruth) {
+                labelview.adopt(ctx.inputs().get<gsl::span<char>>("digitsMCTR"));
+                fillBranch(t.Branch((detStr + "DigitMCTruth").c_str(), &labelview));
```
This is a smart mechanism! I am just wondering if one should not the on exit "delete" these fake vectors by copying an empty `Imp` struct with all nullptrs? Not because of the memory leak of a few bytes but to not have lost dirty pointer?

sawenzel replied:
Yes absolutely. Thanks for pointing out.

sawenzel replied:
So the whole feature was moved into the RootTreeWriter. We could delete the custom treatment here.

shahor02 reviewed (dismissed):
@sawenzel Thanks, very good! Just 1 question below.

sawenzel committed:
TRD: use new zero copy IO of RootTreeWriter

Also remove manipulation of branch lists, somehow
the written TTrees segfault when read back. So for the
moment we have to live with ughly "tmp" branches.

sawenzel committed:
New digit MC label handling in ITSMFT digit + reco

sawenzel committed:
ITSMFT: Zero copy - accumulate digits direcly in DPL resource

sawenzel commented:
Merging. Slight cleanup/beautification will come in a different PR.
